healthcheck_test.go

```diff
@@ -196,3 +230,187 @@ func TestExceptionUnreachable(t *testing.T) {
+
+		httpCli := &dphttp.ClienterMock{DoFunc: func(ctx context.Context, request *http.Request) (*http.Response, error) {
+			if request.Method == "HEAD" {
+				return doIndexDoesNotExist(ctx, request)
+			}
+			return doOkGreen(ctx, request)
+		}}
+
+		cli := elasticsearch.NewClientWithHTTPClient(testUrl, true, httpCli, testIndex)
+
+		// CheckState for test validation
+		checkState := health.NewCheckState(elasticsearch.ServiceName)
+
+		Convey("Checker updates the CheckState to a successful state", func() {
+			cli.Checker(context.Background(), checkState)
+			So(len(httpCli.DoCalls()), ShouldEqual, 2)
+			So(checkState.Status(), ShouldEqual, health.StatusCritical)
+			So(checkState.Message(), ShouldEqual, elasticsearch.ErrorIndexDoesNotExist.Error())
+			So(checkState.StatusCode(), ShouldEqual, 404)
+
+		})
+	})
+}
+
+func TestNoClientIndex(t *testing.T) {
+	Convey("Given that the client has one index and this index does not exists", t, func() {
+
+		httpCli := &dphttp.ClienterMock{DoFunc: func(ctx context.Context, request *http.Request) (*http.Response, error) {
+			if request.Method == "HEAD" {
+				return doNoClientIndex(ctx, request)
+			}
+			return doOkGreen(ctx, request)
+		}}
+
+		cli := elasticsearch.NewClientWithHTTPClient(testUrl, true, httpCli, testNoClientIndex)
+
+		// CheckState for test validation
+		checkState := health.NewCheckState(elasticsearch.ServiceName)
+
+		Convey("Checker updates the CheckState to a successful state", func() {
+			cli.Checker(context.Background(), checkState)
+			So(len(httpCli.DoCalls()), ShouldEqual, 1)
+			So(checkState.Status(), ShouldEqual, health.StatusOK)
+			So(checkState.Message(), ShouldEqual, elasticsearch.MsgHealthy)
+			So(checkState.StatusCode(), ShouldEqual, 200)
+
+		})
+	})
+}
+
+func TestTwoIndexesExist(t *testing.T) {
+	Convey("Given that the client has two indexes and both indexes exist", t, func() {
+
+		httpCli := &dphttp.ClienterMock{DoFunc: func(ctx context.Context, request *http.Request) (*http.Response, error) {
+			if request.Method == "HEAD" {
+				return doIndexExists(ctx, request)
+			}
+			return doOkGreen(ctx, request)
+		}}
+
+		cli := elasticsearch.NewClientWithHTTPClient(testUrl, true, httpCli, testTwoIndexes)
+
+		// CheckState for test validation
+		checkState := health.NewCheckState(elasticsearch.ServiceName)
+
+		Convey("Checker updates the CheckState to a successful state", func() {
+			cli.Checker(context.Background(), checkState)
+			So(len(httpCli.DoCalls()), ShouldEqual, 3)
+			So(checkState.Status(), ShouldEqual, health.StatusOK)
+			So(checkState.Message(), ShouldEqual, elasticsearch.MsgHealthy)
+			So(checkState.StatusCode(), ShouldEqual, 200)
+
+		})
+	})
+}
+
+func TestOneOfTwoIndexesExist(t *testing.T) {
+	Convey("Given that the client has two indexes and both indexes exist", t, func() {
+
+		i := 0
+		httpCli := &dphttp.ClienterMock{DoFunc: func(ctx context.Context, request *http.Request) (*http.Response, error) {
+			i += 1
```

Comment:
This test is completely fine. But I would suggest an improvement (feel free to implement or ignore): I think it would be clearer to check for the URL Path instead of the iteration number (i), something like:
```
if request.Method == "HEAD"{
    if request.Path == "/one" {
        return doIndexExists(ctx, request)
    }
    return doIndexDoesNotExist(ctx, request)
}
return doOkGreen(ctx, request)
```

Reply:
great idea, I had been trying to think of another way to write this test